### ISSUE TITLE
Carry flag not set correctly for cmp

### ISSUE THREAD

**Issue by Benestar** [TUM-LRR/Jasmin#11]
When executing the following code in Jasmin, the carry flag gets set when comparing the value with a register but not when comparing with the same value as a constant.

```
mov al, 1
mov bl, -56
cmp al, bl       <- carry flag set
cmp al, -56     <- carry flag not set
```

However, the carry flag gets set when doing `cmp al, byte -56`.

**Comment by meyerm** [comment]
That's actually a good catch. And the workaround `cmp reg, byte imm` shouldn't even be possible...

**Comment by meyerm** [comment]
Fixed by Patrick's PR #14
